packages/deployments/contracts/contracts/Connext.sol

```diff
@@ -648,6 +675,9 @@ contract Connext is
     if (!isAssetOwnershipRenounced() && !approvedAssets[id]) revert Connext__addLiquidityForRouter_badAsset();
 
     // Transfer funds to coethWithErcTransferact
+    if (_local == address(0) && msg.value != _amount) {
+      revert Connext__addLiquidityForRouter_badAmount();
+    }
```

Comment:
this is not necessary as it is asserted/checked in the `AssetLogic.transferAssetToContract` function

Reply:
Because I updated `transferAssetToContract` function like below due to callback fee.
```
    if (_assetId == address(0)) {
      // When transferring native asset to the contract, always make sure that the
      // asset is properly wrapped
      if (msg.value < _specifiedAmount) revert AssetLogic__transferAssetToContract_notAmount();
      _wrapper.deposit{value: _specifiedAmount}();
      _assetId = address(_wrapper);
    } 
```
So i had to check `msg.value == amount` before calling `transferAssetToContract` function.